### ISSUE TITLE
Blacksmith Forge not smelting some ores correctly.

### ISSUE THREAD

**Issue by Lakue-Bit** [d3athrow/vgstation13#30226]
<bugfix>

#### Description of issue
Forge can't smelt Iron Ore into Metal Sheets
Forge can't smelt Plasma Ore into Plasma Sheets, since Plasma Ore is considered a fuel for the Forge

#### Difference between expected and actual behavior
What you expected: iron and plasma ores turning into sheets so I could make blacksmith autism sword.
What actually happened: No sheets, no blacksmithing autism. Immulsions broken. Day ruined.

#### Steps to reproduce
Throw Plasma Ore in lit/unlit forge, becomes fuel
Throw Iron Ore in lit forge, nothing happens

**Comment by Kurfursten** [comment]
The plasma one is intended I believe

**Comment by Eneocho** [comment]
Intended, yes, but kinda lame.

**Comment by Eneocho** [comment]
Related to #29926.

`rscadd: Blacksmithing forges can now smelt ores, one pile at a time. (except plasma, the fuel)`

**Comment by DeityLink** [comment]
Agreed and easily fixable. Allow plasma sheets for fuel while ores become sheets. Provide hobo shacks witch a few couple plasma sheets.

**Comment by Zth--** [comment]
1071 runtimes on smelting.dm,8 (http://game.ss13.moe/logs/dev/runtime/2021-09-28-runtime.log)

```
[17:26:01] Runtime in smelting.dm,8: Cannot read null.storage
  proc name: checkIngredients (/datum/smelting_recipe/proc/checkIngredients)
  src: Silver (/datum/smelting_recipe/silver)
  call stack:
  Silver (/datum/smelting_recipe/silver): checkIngredients(null)
  Uranium ore (/obj/item/stack/ore/uranium): attempt heating(the forge (/obj/structure/forge), null)
  the forge (/obj/structure/forge): process()
  Objects (/datum/subsystem/obj): fire(0)
  Objects (/datum/subsystem/obj): ignite(0)
  Master (/datum/controller/master): RunQueue()
  Master (/datum/controller/master): Loop()
  Master (/datum/controller/master): StartProcessing()
  ```
https://github.com/vgstation-coders/vgstation13/blob/fdb830f0517174640c27780627c2097149d81507/code/modules/blacksmithing/forge.dm#L136

https://github.com/vgstation-coders/vgstation13/blob/fdb830f0517174640c27780627c2097149d81507/code/modules/mining/smelting.dm#L7-L14
